hycon/interfaces/hercules_interface.py

```diff
@@ -126,47 +129,42 @@ def get_measurements(self, h_dict):
+                measurements["wind_farm"]["power_reference"] = external_signals[
+                    "wind_power_reference"
+                ]
 
-            if "solar_power_reference" in h_dict["external_signals"] and self._has_solar_component:
-                measurements["solar_farm"]["power_reference"] = (
-                    h_dict["external_signals"]["solar_power_reference"]
-                )
+            if "solar_power_reference" in external_signals and self._has_solar_component:
+                measurements["solar_farm"]["power_reference"] = external_signals[
+                    "solar_power_reference"
+                ]
 
             if self._has_battery_component:
-                if "battery_power_reference" in h_dict["external_signals"]:
-                    measurements["battery"]["power_reference"] = (
-                        h_dict["external_signals"]["battery_power_reference"]
-                    )
-
-            if "hydrogen_reference" in h_dict["external_signals"] and self._has_hydrogen_component:
-                measurements["hydrogen"]["power_reference"] = (
-                    h_dict["external_signals"]["hydrogen_reference"]
-                )
-
-            # Grid price information
-            if "lmp_da_00" in h_dict["external_signals"]:
-                measurements["DA_LMP_24hours"] = [
-                    h_dict["external_signals"]["lmp_da_{:02d}".format(h)] for h in range(24)
-                ]
-            if "lmp_da" in h_dict["external_signals"]:
-                measurements["DA_LMP"] = h_dict["external_signals"]["lmp_da"]
-            if "lmp_rt" in h_dict["external_signals"]:
-                measurements["RT_LMP"] = h_dict["external_signals"]["lmp_rt"]
+                if "battery_power_reference" in external_signals:
+                    measurements["battery"]["power_reference"] = external_signals[
+                        "battery_power_reference"
+                    ]
+
+            if "hydrogen_reference" in external_signals and self._has_hydrogen_component:
+                measurements["hydrogen"]["power_reference"] = external_signals["hydrogen_reference"]
+
+            # Grid price information (using pre-computed keys for performance)
+            if "lmp_da_00" in external_signals:
+                measurements["DA_LMP_24hours"] = [external_signals[k] for k in self._lmp_da_keys]
+            if "lmp_da" in external_signals:
+                measurements["DA_LMP"] = external_signals["lmp_da"]
+            if "lmp_rt" in external_signals:
+                measurements["RT_LMP"] = external_signals["lmp_rt"]
 
             # Special handling for forecast elements
-            for k in h_dict["external_signals"].keys():
+            for k, v in external_signals.items():
                 if "forecast" in k:
-                    measurements["forecast"][k] = h_dict["external_signals"][k]
+                    measurements["forecast"][k] = v
```

Comment:
In this case, I think the suggested form probably is a little more pythonic, so I'm ok with this change---but it's still a bit doubtful to me that there is a real performance enhancement

Reply:
reverted